tsdate/prior.py

```diff
@@ -947,48 +972,55 @@ def gamma_cdf(t_set, alpha, beta):
     return np.insert(t_set, 0, 0)
 
 
-def fill_priors(node_parameters, timepoints, ts, Ne, *, prior_distr, progress=False):
+def make_discretised_grid(timepoints, ts):
+    """
+    Create a NodeGridValues object to containing probabilities of nodes existing in
+    a set of specified timeslices, with nodes corresponding to the input tree sequence
     """
-    Take the alpha and beta values from the node_parameters array, which contains
-    one row for each node in the TS (including fixed nodes)
-    and fill out a NodeGridValues object with the prior values from the
-    gamma or lognormal distribution with those parameters.
+    datable_nodes = np.ones(ts.num_nodes, dtype=bool)
+    datable_nodes[ts.samples()] = False
+    datable_nodes = np.where(datable_nodes)[0]
+    return base.NodeGridValues(
+        ts.num_nodes,
+        datable_nodes[np.argsort(ts.tables.nodes.time[datable_nodes])].astype(np.int32),
+        timepoints,
+    )
+
+
+def fill_discretised_grid(grid, node_parameters, progress=False):
+    """
+    Fill out a NodeGridValues object using the alpha and beta values from the
+    node_parameters array, which contains one row for each node in the TS
+    (including fixed nodes). The node_parameters are taken as the params of a
+    gamma or lognormal distribution.
 
     TODO - what if there is an internal fixed node? Should we truncate
     """
-    if prior_distr == "lognorm":
+    if node_parameters.distribution == "lognorm":
         cdf_func = scipy.stats.lognorm.cdf
         main_param = np.sqrt(node_parameters[:, PriorParams.field_index("beta")])
         scale_param = np.exp(node_parameters[:, PriorParams.field_index("alpha")])
-    elif prior_distr == "gamma":
+    elif node_parameters.distribution == "gamma":
         cdf_func = scipy.stats.gamma.cdf
         main_param = node_parameters[:, PriorParams.field_index("alpha")]
         scale_param = 1 / node_parameters[:, PriorParams.field_index("beta")]
     else:
         raise ValueError("prior distribution must be lognorm or gamma")
 
-    datable_nodes = np.ones(ts.num_nodes, dtype=bool)
-    datable_nodes[ts.samples()] = False
-    datable_nodes = np.where(datable_nodes)[0]
-    prior_times = base.NodeGridValues(
-        ts.num_nodes,
-        datable_nodes[np.argsort(ts.tables.nodes.time[datable_nodes])].astype(np.int32),
-        timepoints,
-    )
-
     # TO DO - this can probably be done in an single numpy step rather than a for loop
     for node in tqdm(
-        datable_nodes, desc="Assign Prior to Each Node", disable=not progress
+        grid.nonfixed_nodes, desc="Assign Prior to Each Node", disable=not progress
     ):
         with np.errstate(divide="ignore", invalid="ignore"):
-            prior_node = cdf_func(timepoints, main_param[node], scale=scale_param[node])
+            prior_node = cdf_func(
+                grid.timepoints, main_param[node], scale=scale_param[node]
```

Comment:
If we quote the params in coalescent scaled time here, we need `grid.timepoints` to be in coalescent time too. It seems to me that we would then need to apply a conversion from coalescent time to "true" time, using whatever function was passed in from the coalescent point distribution